src/mqt/qecc/mod2.py

```
@@ -0,0 +1,112 @@
+# Licensed under the MIT License
+
+"""Utilities for binary linear algebra (mod 2) to replace dependency on ldpc.mod2."""
+
+from __future__ import annotations
+
+import numpy as np
+
+
+def row_echelon(
+    matrix: np.ndarray,
+    full: bool = False,
```

Ruff (0.15.20)

[warning] Boolean-typed positional argument in function definition

(FBT001)